pkg/handler/crud.go

```diff
@@ -127,12 +127,56 @@ func (c *crud) CreateFlag(params flag.CreateFlagParams) middleware.Responder {
 		}
 		f.Key = key
 	}
-	err := getDB().Create(f).Error
-	if err != nil {
+
+	tx := getDB().Begin()
+
+	if err := tx.Create(f).Error; err != nil {
+		tx.Rollback()
 		return flag.NewCreateFlagDefault(500).WithPayload(
 			ErrorMessage("cannot create flag. %s", err))
 	}
 
+	if params.Body.Template == "simple" {
```

Comment:
I was hoping there might be a better way to use a constant here, but got a bit overwhelmed when going down that route and figuring out how it'd make sense w/ swagger/etc.

Reply:
Agreed, I think swagger is more for schema, inside backend, we can use constants as validation. Actually, it's also a good opportunity to do some refactoring, this function is too long. We can have a file `crud_flag_creation.go` in the same handler folder, and dedicates the creation by template logic there. Also the template name `simple`, variant `on` key can be defined as constants.

For the naming of template, how about we start with something explicit first `simple_boolean_flag` instead of `simple`? It's easier to extend that to `simple_ab_experiment` and `simple_dynamic_configuration` in the future.